pkg/smart_wallet/factory_calldata.go

```diff
@@ -0,0 +1,70 @@
+package smart_wallet
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/shopspring/decimal"
+)
+
+func GetFactoryCallData(smartWalletConfig Config, ownerAddress common.Address, index decimal.Decimal) ([]byte, error) {
+	var initCode []byte
+	var err error
+
+	switch typ := *smartWalletConfig.Type; typ {
+	case SimpleAccountType:
+		return nil, fmt.Errorf("%w: %s", ErrSmartWalletNotSupported, typ)
+	case BiconomyType: // not tested
+		initCode, err = GetBiconomyFactoryCallData(ownerAddress, index, smartWalletConfig.ECDSAValidator)
+	case KernelType:
+		initCode, err = GetKernelFactoryCallData(ownerAddress, index, smartWalletConfig.Logic, smartWalletConfig.ECDSAValidator)
+	default:
+		return nil, fmt.Errorf("unknown smart wallet type: %s", typ)
+	}
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to get init code: %w", err)
+	}
+
+	return initCode, nil
+}
+
+// getKernelInitData returns the calldata needed call the factory
```

Comment:
Godoc states the function is private but it's no longer the case.
```suggestion
// GetKernelInitData returns the calldata needed call the factory
```

Reply:
6982100528d7e2cfb6f19798fbceeaf3e24497b1